### PR TITLE
Next batch of Doxygen updates.

### PR DESCRIPTION
**Pull request by gheber**
New stuff:
- Meta-documentation: documentation about documentation: https://docs.hdfgroup.org/hdf5/develop/_about.html
- More RFCs: https://docs.hdfgroup.org/hdf5/develop/_r_f_c.html
- Great technical notes: https://docs.hdfgroup.org/hdf5/develop/_t_n.html
- History of the file format spec: https://docs.hdfgroup.org/hdf5/develop/_s_p_e_c.html
- Full-text search: https://docs.hdfgroup.org/hdf5/develop/_f_t_s.html

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1192,15 +1192,14 @@ if test "X$HDF5_DOXYGEN" = "Xyes"; then
   AC_SUBST([DOXYGEN_SERVER_BASED_SEARCH])
   AC_SUBST([DOXYGEN_EXTERNAL_SEARCH])
   AC_SUBST([DOXYGEN_SEARCHENGINE_URL])
-  AC_SUBST([DOXYGEN_STRIP_FROM_PATH])
```
Comment (byrnHDF) [inline configure.ac:1195]: CMake change too?

Comment (gheber) [reply configure.ac:1195]: Will do.

**Review by byrnHDF — APPROVED** [review]

File: configure.ac (hunk 1 above)
Comment (gheber) [reply configure.ac:1195]: Now I don't remember why I got rid of it.

Comment (gheber) [reply configure.ac:1195]: Restored.

**Review by byrnHDF — CHANGES_REQUESTED** [review]
CMake still needs update

**Comment by gheber** [comment]
Fixed CMake.

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**